cosmos/dataset.py

```diff
@@ -15,8 +15,10 @@ def __init__(self, id: str, *args: Tuple[Any], **kwargs: str):
             self.id = id
             logger.warning("Datasets are not supported in Airflow < 2.5.0")
 
-        def __eq__(self, other: "Dataset") -> bool:
-            return bool(self.id == other.id)
+        def __eq__(self, other: object) -> bool:
+            if not isinstance(other, Dataset):
+                return NotImplemented
+            return bool(self.id == other.id)  # type: ignore[attr-defined]
```

Comment:
see if swapping the cases fixes this

Reply:
it doesn't, seems like ignoring is the least invasive solution rn